### DISCUSSION TITLE
Cannot define new `DELETE /users/me` endpoint

### DISCUSSION THREAD

**Discussion by MatthewScholefield** [fastapi-users/fastapi-users#444] [Q&A] [answered]
In my API, I would like to support a user being able to delete their own account (like most sites that have a delete button within their account page). To do this, I figured I would add a new route:

```python
@app.delete("/users/me", status_code=204)
async def delete_me(user: UserDB = Depends(fastapi_users.get_current_user)):
    await user_db.delete(user)
    return None
```

However, this doesn't work because `DELETE /users/me` always returns `403` (forbidden), most likely because it is being swallowed by the existing super-user route under `DELETE /users/{id}`.

Any idea if there's a way to override a FastAPI route or would it be worth having an option to customize which routes get applied?

Overall, it's not a big deal though because it just means my API's naming is a bit more clunky with something like `POST /delete-account` or using some synonym like `DELETE /accounts/me`.

**Comment by frankie567** [comment, marked as answer]
You're right, the superuser route `DELETE /{id}` is taking precedence there.

I've played around  a bit and found out that if we suffix the superuser routes with the convertor`:uuid` (a [Starlette feature](https://www.starlette.io/routing/#path-parameters) not documented in FastAPI), it lets pass `/me` route that would be defined after.

Sounds a quick, easy and relevant fix. I'll implement this quickly!

**Reply by frankie567** [reply]
Well, "quickly" turned out to be a month... Sorry guys 😔

See #475

**Reply by frankie567** [reply]
Fixed in v5.0.1 🎉

**Reply by bkis** [reply]
But does this work for people using `fastapi-users-beanie` where BSON `ObjectId`s are used internally (and not UUIDs)?

EDIT: I just tried. It doesn't :(

**Reply by frankie567** [reply]
This is an old discussion, so it's not true anymore that the routes have the `:uuid` convertor.

I tried and it works correctly with Beanie and ObjectId, which is the default on the provided example: https://github.com/fastapi-users/fastapi-users/tree/master/examples/beanie

**Reply by SnoozeFreddo** [reply]
It doesn't work with sqlalchemy tho.

**Reply by bkis** [reply]
@SnoozeFreddo I don't know if you already checked this, but for me the pitfall was that I let FastAPI-Users register its routes _before_ I registered the routes of my own application. I had to make sure it happens the other way around, so a `DELETE` on `/users/me` _first_ matches my custom endpoint if possible and _then_ falls through to the `/users/<id>` endpoint of FastAPI-Users.

**Reply by SnoozeFreddo** [reply]
Makes sense, u mean that you use app.include_router before the auth_router and you do not define your custom endpoind on the auth_router ( get_auth_router) but a custom one?

My super quick workaround was to have it on a completetly different name

`users_router = fastapi_users.get_users_router(...)`

````
@users_router.delete('/delete/me',status_code=status.HTTP_204_NO_CONTENT)
async def delete_my_account(
        request: Request,
    user_manager: UserManager = Depends(get_user_manager),
    user: User = Depends(current_active_verified_user),

):
    await user_manager.delete(user, request)
    return 204
````

**Reply by bkis** [reply]
Exactly, I have a couple of different routers, including a custom `users` router with multiple endpoints (DELETE `/users/me`, GET `/users` for paginated users search, etc.). Then I register them in order.

```py
# register my app's routers
for router in _some_func_that_returns_all_my_routers():
    app.include_router(router)
# register FastAPI-Users' routers
# ...
```

Your solution looks very intuitive to me, I might have even tried the same back then, but when declaring your `/users/me` endpoint on the `users_router` you get from FastAPI-Users, the internal `/users/<id>` endpoint is already there and takes precedence.
